### ISSUE TITLE
TypeError on submitting an executable with metadata set

### ISSUE THREAD

**Issue by kam193** [CybercentreCanada/assemblyline#244]
**Describe the bug**
When submitting [this file](https://www.virustotal.com/gui/file/a96a6b9ef81a1f7ad1b29e8b50c6dee1ad65df1006f72c6b7a65f382a9fb12c1/detection) to my AL instance, with a manually set metadata in the submission form, the submitting process stops on 100%. Looking into network connections, I've found an error returned from the API:

```
 File "/var/lib/assemblyline/.local/lib/python3.11/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
 File "/var/lib/assemblyline/.local/lib/python3.11/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/assemblyline/.local/lib/python3.11/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^  
File "/var/lib/assemblyline/.local/lib/python3.11/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/assemblyline/.local/lib/python3.11/site-packages/assemblyline_ui/api/base.py", line 235, in base
    return func(*args, **kwargs) 
          ^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/assemblyline/.local/lib/python3.11/site-packages/assemblyline_ui/api/v4/ui.py", line 170, in flowjs_upload_chunk
    t.write(cache.get(chunk_name))
TypeError: a bytes-like object is required, not 'NoneType'
```

There are no more details in the logs. The same file sent without metadata works fine, as well as other files sent with metadata. After I've sent the file without metadata, uploading again with metadata works well (...or maybe just because I was following logs and the container felt being observed...? :)). So far, I didn't find any other file causing the issue - it looks it may not be easy to reproduce

In addition, it looks like the `ui` container stops responding timely after 1-2 tries with this file (nginx reports upstream timeout on most of the requests, nothing significant in the `ui` logs; container restart helps).

**To Reproduce**
1. Have an AL instance with some metadata configured to optionally set during the submission. In my case, there are two optional strings.
3. Go to 'Submit'
2. Set a value to one of the metadata fields.
4. Select [file](https://www.virustotal.com/gui/file/a96a6b9ef81a1f7ad1b29e8b50c6dee1ad65df1006f72c6b7a65f382a9fb12c1/detection) to upload.
5. Observe freeze on 100% uploading progress.
6. Find error in Dev tools (if activated earlier) on logs of the `ui` container.

**Expected behavior**
* File is correctly uploaded.
* Error from API is somehow presented in the UI.

**Screenshots**
![Screen](https://github.com/user-attachments/assets/6bf6cd9a-2345-42da-8567-3bbdd17f2119)

**Environment (please complete the following information if pertinent):**
 - Assemblyline Version: 4.5.0.40
 - Browser: Firefox
 - Deployed as Docker Compose

**Additional context**

**Comment by cccs-rs** [comment]
Looks like this particular error (and others like it that yield a frozen upload progress) are related by internal server errors (HTTP 500).

As mentioned, it is hard to reproduce this by hand but we can update the frontend code to propagate these messages forward to the user to indicate there was a problem and they should try re-uploading the file.

**Comment by kam193** [comment]
Thanks! On one hand, it may be tricky to reproduce... on the other, I've just got it again with entirely different windows executable, and once more looks like the "solution" is to try without setting metadata. Although, not fully - this time after trying without metadata, AL got frozen as well, and the `ui` container started spamming logs with:

```
{"@timestamp": "2024-08-09 19:34:28,830", "event": { "module": "assemblyline", "dataset": "assemblyline.datastore" }, "host": { "ip": "172.16.0.7", "hostname": "cebc97db5b42" }, "log": { "level": "INFO", "logger": "assemblyline.datastore" }, "process": { "pid": "3173" }, "message": "Retrying save or freshen due to version conflict: ConflictError(409, 'version_conflict_engine_exception', '[0afdfd1233929dce9fafcfa12827d46c67fc8f273089ced2177cd40ab262736b]: version conflict, required seqNo [1217], primary term [2]. but no document was found')"}
```

By "spamming" I mean it sends even multiple log entries pro second and does not want to stop without container restart :sweat_smile:

**Comment by cccs-rs** [comment]
This should be resolved in the [4.5.0.41](https://github.com/CybercentreCanada/assemblyline/releases/tag/v4.5.0.stable41) release of Assemblyline
